src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -253,12 +253,17 @@ private void RemoveDocumentFromWorkspace_NoLock(MetadataAsSourceGeneratedFileInf
             }
         }
 
-        private void InitializeWorkspace(Project project)
+        private void InitializeWorkspace(Project project, OptionSet option)
```

Comment:
Basically, you'll provide an implementation of IOptionProvider that doesn't use the serializers.